File: src/main/java/net/minestom/server/command/builder/suggestion/SuggestionEntry.java
package net.minestom.server.command.builder.suggestion;

import net.kyori.adventure.text.Component;
import org.jetbrains.annotations.Nullable;

import java.util.Objects;

public class SuggestionEntry {
    private final String entry;
    private final Component tooltip;

    public SuggestionEntry(String entry, @Nullable Component tooltip) {
        this.entry = entry;
        this.tooltip = tooltip;
    }

    public SuggestionEntry(String entry) {
        this(entry, null);
    }

    public String getEntry() {
        return entry;
    }

    public @Nullable Component getTooltip() {
        return tooltip;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof SuggestionEntry that)) return false;
        return getEntry().equals(that.getEntry()) && Objects.equals(getTooltip(), that.getTooltip());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getEntry(), getTooltip());
    }
}

File: src/main/java/net/minestom/server/command/builder/suggestion/SuggestionEntry.kt
package net.minestom.server.command.builder.suggestion

import net.kyori.adventure.text.Component
import net.minestom.server.command.builder.suggestion.SuggestionEntry
import net.minestom.server.command.builder.CommandContext

class SuggestionEntry @JvmOverloads constructor(val entry: String, val tooltip: Component? = null)